ztools/ztls/handshake_client.go

```diff
@@ -580,7 +588,6 @@ func (hs *clientHandshakeState) doFullHandshake() error {
 		sr = hs.serverHello.random
 	}
 
-	hs.masterSecret = masterFromPreMasterSecret(c.vers, hs.suite, preMasterSecret, cr, sr)
```

Comment:
The master secret calculation needs to be moved here to account for extended random.